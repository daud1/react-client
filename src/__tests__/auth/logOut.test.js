import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LogOutwithRouter, { LogOut } from '../../components/auth/logOut.component';
import moxios from 'moxios';


describe ('LogOut Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<LogOutwithRouter />);
        expect(wrapper.exists(<button></button>)).toBe(true)
    });
    
    it('matches snapshot', () => {
        const wrapper = shallow(<LogOutwithRouter />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<LogOut />);
        // wrapp = wrapper.find('button');
        console.log(wrapper.debug());
        // wrapper.find("button").simulate('click', { preventDefault () {} });

        moxios.stubRequest('http://localhost:5000/auth/logout', {
            status: 200,
            response: {
                'MSG': 'Successfully Logged Out!',
            }
        });
        
        moxios.wait(() => {
            expect(wrapper.html()).toContain('Successfully Logged Out');
            done();
        });
    });
})