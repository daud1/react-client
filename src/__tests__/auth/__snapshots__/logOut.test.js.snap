// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogOut Component matches snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
